pkg/buffer/buffer.go

```diff
@@ -102,6 +106,7 @@ type Stats struct {
 // BufferOptions provides configuration options for the buffer
 type Options struct {
 	MaxBitRate uint64
+	Logger     logr.Logger
```

Comment:
And just get rid of all these embedded ones.

Reply:
Removed